stratisd_proc_macros/src/lib.rs

```diff
@@ -193,59 +193,45 @@ fn wrap_method(f: &mut ImplItemMethod) {
 /// Get the pool available action state level at which a pool operation ceases to be
 /// accepted.
 fn get_attr_level(attrs: &mut Vec<Attribute>) -> Option<Ident> {
-    let mut return_value = None;
-    let mut index = None;
-    for (i, attr) in attrs.iter().enumerate() {
-        if let Meta::List(MetaList {
-            ref path,
-            ref nested,
-            ..
-        }) = attr
-            .parse_meta()
-            .unwrap_or_else(|_| panic!("Attribute {attr:?} cannot be parsed"))
-        {
-            if path
-                == &parse_str("pool_mutating_action").expect("pool_mutating_action is valid path")
-            {
-                for nested_meta in nested.iter() {
-                    if let NestedMeta::Lit(Lit::Str(litstr)) = nested_meta {
-                        index = Some(i);
-                        return_value =
-                            Some(parse_str::<Ident>(&litstr.value()).unwrap_or_else(|_| {
-                                panic!("{} is not a valid identifier", litstr.value())
-                            }));
-                    } else {
-                        panic!("pool_mutating_action attribute must be in form #[pool_mutating_action(\"REJECTION LEVEL\")]");
-                    }
+    if let Some((index, return_value)) = attrs
+        .iter()
+        .enumerate()
+        .filter(|(_, a)| a.path().is_ident("pool_mutating_action"))
+        .map(|(i, a)| {
+            if let Meta::List(MetaList { ref tokens, .. }) = a.meta {
+                if let TokenTree::Literal(litstr) = tokens.clone().into_iter().next().expect("stratisd code always supplies string literal argument to pool_mutating_actions") {
```

Comment:
I think this can be further destructured using [this enum](https://docs.rs/syn/latest/syn/enum.Lit.html). It will allow you to avoid converting to a String and then trimming the quotation marks, instead just getting the String value.

Reply:
litstr type is  TokenTree::Literal, which is a struct with private fields and a rather awkward set of methods in syn2. I think that syn2  probably expects that the normal way to encode that argument value is without the '"' in the first place, as a plain Ident, but I don't want to hit every spot where the ```pool_mutating_action``` annotation is used to change that, so I left it alone.

Reply:
Ah, I see. I think changing this to use an ident would be better after looking at the types. Would you be willing to do that? I think it would generally be better from the standpoint of upgrading to the new API. I'll also look into the recommended way of doing this because I have some other crates I maintain that need updates.

Reply:
I would, but I'ld like to postpone that until we start the next release. I'll file an issue.

Reply:
Okay sounds good!